### PR TITLE
Landing page and platform merge

### PR DESCRIPTION
**Pull request by aapeliv**
First cut at merging landing site and app.

TODO:
* ~~Navbar for signed out users~~
* ~~Allowing signed out users to view markdown pages~~
* ~~SEO tags (meta tags, etc)~~
* ~~Main home page~~
* ~~Redirects~~
* ~~Team page~~
* ~~Missing "scroll to learn more"~~

**Web frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
Old:
Screen Shot 2022-01-15 at 3 16 26 pm


New:
Screen Shot 2022-01-15 at 3 16 32 pm


Old:
Screen Shot 2022-01-15 at 3 16 48 pm

New:
Screen Shot 2022-01-15 at 3 16 52 pm


Old:
Screen Shot 2022-01-15 at 3 18 39 pm

New:
Screen Shot 2022-01-15 at 3 24 23 pm

**Comment by aapeliv** [comment]
Logged out menus:
Screen Shot 2022-01-15 at 4 41 20 pm
Screen Shot 2022-01-15 at 4 41 15 pm
Screen Shot 2022-01-15 at 4 41 02 pm

**Comment by aapeliv** [comment]
Team page:

old:
Screen Shot 2022-01-15 at 5 56 09 pm

new:
Screen Shot 2022-01-15 at 5 56 16 pm

**Comment by aapeliv** [comment]
@lucaslcode: I can't get the meta tags to work properly. They're all just the same "Couchers.org Beta" in the vercel preview and when doing `next build && next start` locally.

They are being updated in the DOM, but not in the generated HTML that's sent back from the server.

**Comment by aapeliv** [comment]
OK, these pages are straight up just not being SSRd, I don't know what I'm doing wrong or if I'm just very confused here!

**Review by lucaslcode — COMMENTED** [review]
Great work! I will look properly later. Your `getStaticPaths` needs to be async and await the slugs function.

**Review by lucaslcode — COMMENTED** [review]
Some general stuff:
- There's a bug in AppRoutes so these pages are statically rendering a loader instead of the page. I think line 79 `!isMounted || (isPrivate && !isAuthenticated)` should be `(!isMounted && isPrivate) || (isPrivate && !isAuthenticated)`
- Are you using markdown-it instead of our markdown component because ours doesn't work with ssr? Maybe we should replace that one with markdown-it.
- The tag colours look weird, but I can't offer any suggestions (except for maybe removing them completely?)

**Comment by aapeliv** [comment]
> There's a bug in AppRoutes so these pages are statically rendering a loader instead of the page. I think line 79 `!isMounted || (isPrivate && !isAuthenticated)` should be `(!isMounted && isPrivate) || (isPrivate && !isAuthenticated)`

It works well (on these pages) and is server side rendered (turning off JS) if I just replace that logic with `false` (aka never show the spinner). It seems to be something to do with `isMounted`, I don't know what that does.

**Comment by aapeliv** [comment]
>     * Are you using markdown-it instead of our markdown component because ours doesn't work with ssr? Maybe we should replace that one with markdown-it.

I'm just using it because it was used on Vue and I wanted everything to be as similar as possible.

>     * The tag colours look weird, but I can't offer any suggestions (except for maybe removing them completely?)

I think we can leave it for now. The tag colors match the boxes on the current landing page when you scroll down.

**Comment by aapeliv** [comment]
@lucaslcode: I've implemented the landing page now. It works as follows:

1. if you're logged in, `/` shows you the dashboard
2. if you're logged out, `/` shows the landing page as currently on couchers.org

I'm out of juice on this at the moment. I know it's a pretty crap PR in terms of code quality...

Random stuff:
* ~~I don't know why Vercel is failing, for all I can see it should work~~
* It's missing a little "scroll down for more" arrow on the landing page
* I tweaked some of the login/signup CSS. It seems it's quite a bit of a mess at the moment (and I doubt I made it any better).
* The landing page AppRoute thing is a mess: if logged in it should be `variant = "standard"`, but if logged out, it should be `full-screen`. This is the `currentVariant` mess.
* This does not follow the UI/UX designs for the landing page. I don't think they're quite ready and thought out yet, and I don't have the skill/patience/juice to build that. So I ported over our existing landing page and made-do. This is (in my opinion) a small regression visually and in UI/UX, but I think it's worth it and we can tweak it further later. I just want to get rid of the awful vue landing site.
* SSR still doesn't work, it's something to do with `isMounted`, if I just comment that out it works fine (but flashes the wrong type of page at times)
* I changed the navbar to be `fixed` and added padding. This gives us pages that are the right height (e.g. login/signup) with just `100vh`, and allows the landing page above-fold to just be `100vh` and then have a nice page below.
* The landing page isn't set up fully with translations yet, I ran out of juice. But it's not a huge thing since I added the `landing` translation component. The translation keys are also a tad bit of a mess in like duplicates between `auth`, etc.
* I bastardized `BasicForm` to put it on the landing page. After you fill it in it'll redirect to `/signup` (but it flashes first). If you've already filled it in it and you goon the landing page again it shows a button to finish signing up.

File: app/web/components/AppRoute.tsx [hunk 1]
```diff
@@ -43,7 +44,7 @@ export const useAppRouteStyles = makeStyles((theme) => ({
 interface AppRouteProps {
   isPrivate: boolean;
   noFooter?: boolean;
-  variant?: "standard" | "full-screen" | "full-width";
+  variant?: "standard" | "full-screen" | "full-width" | "landing";
```
Comment (lucaslcode) [inline app/web/components/AppRoute.tsx]: Check

File: app/web/components/IconText.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ const useStyles = makeStyles((theme) => ({
 
 interface IconTextProps {
   icon: OverridableComponent<SvgIconTypeMap<unknown, "svg">>;
-  text: string;
+  text: ReactNode;
```
Comment (lucaslcode) [inline app/web/components/IconText.tsx:21]: Don't include textbody for type=string

File: app/web/components/Navigation/LoggedInMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        ref={menuRef}
+      >
+        <MenuIcon />
+        <Avatar user={user} className={classes.avatar} isProfileLink={false} />
+      </Button>
+      <Menu
+        id="navigation-menu"
+        open={menuOpen}
+        anchorEl={menuRef.current}
+        onClose={() => setMenuOpen(() => false)}
```
Comment (lucaslcode) [inline app/web/components/Navigation/LoggedInMenu.tsx]: (false)

File: app/web/components/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -128,8 +131,62 @@ const navMenu = (data: MenuGenData): Array<MenuItemProps> => [
...
+    name: "FAQ",
+    route: "/faq",
+  },
+  {
+    name: "The Team",
+    route: "/team",
+  },
+];
+
+const loggedOutDrawerMenu = (data: MenuGenData): Array<MenuItemProps> => [
```
Comment (lucaslcode) [inline app/web/components/Navigation/Navigation.tsx]: rename data

File: app/web/components/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -378,7 +432,7 @@ export default function Navigation() {
 
   return (
     <AppBar
-      position="sticky"
+      position="fixed"
```
Comment (lucaslcode) [inline app/web/components/Navigation/Navigation.tsx]: check margins

File: app/web/features/Index.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { useAuthContext } from "./auth/AuthProvider";
+import Dashboard from "./dashboard/Dashboard";
+import LandingPage from "./landing/LandingPage";
+
+export default function Index() {
+  const { authState } = useAuthContext();
+  const isAuthenticated = authState.authenticated;
+
+  if (isAuthenticated) {
```
Comment (lucaslcode) [inline app/web/features/Index.tsx]: Change to a rewrite

File: app/web/features/auth/locales/en.json [hunk 7]
```diff
@@ -156,5 +156,7 @@
     },
     "notification_settings": {
         "title": "Notification settings"
-    }
+    },
+    "sign_in": "Sign in",
+    "create_account": "Create an account"
```
Comment (lucaslcode) [inline app/web/features/auth/locales/en.json]: rename? global?

File: app/web/components/AppRoute.tsx [hunk 8]
```diff
@@ -19,6 +19,7 @@ export const useAppRouteStyles = makeStyles((theme) => ({
   },
   nonFullScreenStyles: {
     height: "100%",
+    padding: "64px 0 0 0",
```
Comment (lucaslcode) [inline app/web/components/AppRoute.tsx]: Make navbar effect clearer/use variable

File: app/web/features/auth/useAuthStyles.ts [hunk 9]
```diff
@@ -94,31 +98,30 @@ const useAuthStyles = makeStyles((theme) => ({
...
+      margin: theme.spacing(0),
     },
   },
   introduction: {
     flexShrink: 0,
     color: theme.palette.common.white,
     flexDirection: "column",
     display: "flex",
-    justifyContent: "flex-end",
+    // justifyContent: "flex-end",
```
Comment (lucaslcode) [inline app/web/features/auth/useAuthStyles.ts]: rm

File: app/web/features/landing/LandingPage.tsx [hunk 10]
```diff
@@ -0,0 +1,521 @@
...
+  section: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
+    margin: theme.spacing(3),
+    marginTop: theme.spacing(12),
+    [theme.breakpoints.down("md")]: {
+      marginTop: theme.spacing(6),
+    },
+  },
```
Comment (lucaslcode) [inline app/web/features/landing/LandingPage.tsx]: MUI?

File: app/web/features/markdown/MarkdownPage.tsx [hunk 11]
```diff
@@ -0,0 +1,182 @@
+import { Breadcrumbs, Link, makeStyles, Typography } from "@material-ui/core";
+import HtmlMeta from "components/HtmlMeta";
+import markdown from "markdown-it";
+
+const mkd = new markdown();
```
Comment (lucaslcode) [inline app/web/features/markdown/MarkdownPage.tsx:11]: Make issue to replace other md renderer or do here if easy

File: app/web/features/markdown/MarkdownPage.tsx [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+  slug: Array<string>;
+  frontmatter: MarkdownPageFrontmatter;
+  content: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
```
Comment (lucaslcode) [inline app/web/features/markdown/MarkdownPage.tsx]: container mui

File: app/web/features/team/Team.tsx [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+import Link from "next/link";
+import { contributeRoute } from "routes";
+
+import TeamData from "./team.json";
+
+const useStyles = makeStyles((theme) => ({
+  tightSection: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
```
Comment (lucaslcode) [inline app/web/features/team/Team.tsx]: mui

File: app/web/features/team/Team.tsx [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+  cardContent: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    marginLeft: theme.spacing(2),
+    marginRight: theme.spacing(2),
+    flex: "1 0 auto",
+  },
+  avatar: {
+    width: "96px",
+    height: "96px",
```
Comment (lucaslcode) [inline app/web/features/team/Team.tsx]: no magic

File: app/web/features/team/team.json [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+  {
+    "name": "Dominic Nayan",
+    "role": "Head of UX",
+    "location": "Ensenada, Mexico",
+    "img": "/img/team/head-dom.jpg"
+  },
+  {
+    "name": "Lucas Levin",
+    "role": "Head of Frontend (web)",
+    "location": "London, United Kingdom",
```
Comment (lucaslcode) [inline app/web/features/team/team.json]: change

**Review by lucaslcode — COMMENTED** [review]
This is notes for myself

**Comment by lucaslcode** [comment]
This is basically done but needs some checking

**Comment by lucaslcode** [comment]
Note that the landing page/dashbiard redirect middleware doesn't work on preview, because the cookie domain is .couchershq.org, so web browsers don't include it in the request.

**Review by lucaslcode — APPROVED** [review]
This should be good once all checks are green

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Wait the home page still isn't pre-rendered

**Review by lucaslcode — APPROVED** [review]
Okay NOW when checks are green

**Pull request merged**